src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java

```diff
@@ -191,12 +191,35 @@ private void commaSeparatedEntryTransformer(String ocrNameWithCommaSeparatedValu
         exceptionRecordToCcdFieldsMap.put(OcrFieldName.AUTHORISATION_SIGNED_BY, "authorisationSignedBy");
         exceptionRecordToCcdFieldsMap.put(OcrFieldName.AUTHORISATION_SOLICITOR_POSITION, "authorisation2b");
 
+        exceptionRecordToCcdFieldsMap.put(OcrFieldName.RESPONDENT_SOLICITOR_FIRM, "rSolicitorFirm");
+        exceptionRecordToCcdFieldsMap.put(OcrFieldName.RESPONDENT_SOLICITOR_NAME, "rSolicitorName");
+
         return exceptionRecordToCcdFieldsMap;
     }
 
-    private void applyMappingsForAddress(
-            String prefix, List<OcrDataField> ocrDataFields, Map<String, Object> modifiedMap) {
-        applyMappings(prefix, prefix + "Address", ocrDataFields, modifiedMap);
+    private void applyMappingsForAddressExtended(
+            String prefix, String parentField, List<OcrDataField> ocrDataFields, Map<String, Object> modifiedMap) {
+        String nestedFieldPrefix = StringUtils.capitalize(prefix + "Address");
+
+        addMappingsTo(
+            parentField,
+            extendedAddress(nestedFieldPrefix),
+            modifiedMap,
+            ocrDataFields
+        );
+    }
+
+    private static ImmutableMap<String, String> extendedAddress(String prefix) {
```

Comment:
I'd call this something like "getExtendedAddressMapForPrefix"

Reply:
removed. All in bsp-common-lib now